.jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat

```diff
@@ -0,0 +1,52 @@
+if exist "%TMP_DIR%/ci_scripts/pytorch_env_restore.bat" (
+    call %TMP_DIR%/ci_scripts/pytorch_env_restore.bat
+    exit /b 0
+)
+
+set PATH=C:\Program Files\CMake\bin;C:\Program Files\7-Zip;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\Amazon\AWSCLI;%PATH%
+
+:: Install Miniconda3
+if "%BUILD_ENVIRONMENT%"=="" (
+    set CONDA_PARENT_DIR=%CD%
+) else (
+    set CONDA_PARENT_DIR=C:\Jenkins
+)
+if NOT "%BUILD_ENVIRONMENT%"=="" (
+    IF EXIST %CONDA_PARENT_DIR%\Miniconda3 ( rd /s /q %CONDA_PARENT_DIR%\Miniconda3 )
+    curl https://repo.continuum.io/miniconda/Miniconda3-latest-Windows-x86_64.exe --output %TMP_DIR_WIN%\Miniconda3-latest-Windows-x86_64.exe
+    %TMP_DIR_WIN%\Miniconda3-latest-Windows-x86_64.exe /InstallationType=JustMe /RegisterPython=0 /S /AddToPath=0 /D=%CONDA_PARENT_DIR%\Miniconda3
+)
+call %CONDA_PARENT_DIR%\Miniconda3\Scripts\activate.bat %CONDA_PARENT_DIR%\Miniconda3
+if NOT "%BUILD_ENVIRONMENT%"=="" (
+    :: We have to pin Python version to 3.6.7, until mkl supports Python 3.7
+    call conda install -y -q python=3.6.7 numpy mkl cffi pyyaml boto3 protobuf numba
+)
+pip install -q ninja future hypothesis "librosa>=0.6.2" psutil
+
+pushd
```

Comment:
Wait! It should be `pushd .` or `pushd %CD%`.

Reply:
> It should be `pushd .` or `pushd %CD%`.

I [checked the API first](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/pushd) :) - the default is the current directory.  I made sure the Windows builds pass as well.

Reply:
Yes, but I tried it locally and found that it was not true.
```cmd
C:\Users\peter>pushd

C:\Users\peter>cd c:\

c:\>popd

c:\>cd C:\Users\peter

C:\Users\peter>pushd .

C:\Users\peter>cd c:\

c:\>popd

C:\Users\peter>
```

Reply:
The API of MSDN on this function is confusing. Here is a clearer one: https://ss64.com/nt/pushd.html

Reply:
Wow, you're right.

Experimentally I also found that `pushd ..\..` doesn't work.

Reply:
`pushd ..\..` actually works for me. But for safety considerations, using absolute paths are better. I also found from this [page](https://ss64.com/nt/cd.html) that we can use `cd /d [SOME PATH]`.

Reply:
Ah, *facepalm*.  For some reason I was thinking it pushed specified directory onto the stack.  But rather, `pushd` adds the current directory to the stack before changing to the specified directory.